src/Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -230,7 +230,18 @@ private async Task OnProcessBlockAsync(Block block, CancellationToken cancellati
                         }
 
                         this.walletTip = next;
```

Comment:
So my understanding of this is as follows:

- We have determined that the `newTip` is ahead of the `walletTip`
- We loop until the `walletTip` is equal to `newTip` (or a reorg hits midway)
- We increment `walletTip` each iteration, before processing the block itself that will throw the exception due to a reorg
- But this is okay, because in the next call to `OnProcessBlockAsync`, it will be determined that the wallet tip is on the wrong chain and be rewound accordingly

What I'm wondering is, can the `ProcessBlock()` in line 262 experience the same exception if there is a reorg between `newTip` getting assigned, and the reorg check being performed inside `walletManager.ProcessBlock()`? I suppose it very much depends on the circumstances under which the ChainIndexer gets updated.